server.cc

```diff
@@ -105,137 +105,142 @@ int CheckpointProducer::StartStreaming(
     std::bind(&CheckpointProducer::ReplicationDone, this, _1, _2); 
 
   // Staring producer
-  auto rc = producer_->Start(client_ip_, req.consumer_port, req.max_num_of_threads,
+  RepStatus rc = producer_->Start(client_ip_, req.consumer_port, req.max_num_of_threads,
                               parallelism_, timeout_msec, done_cb);
-  if (rc) {
-    logger->Log(LogLevel::ERROR, FormatString("Producer::Start failed\n"));
-    return -1;
+  if (!rc.IsOk()) {
+    logger->Log(Severity::ERROR, FormatString("Producer::Start failed\n"));
+    DestroyCheckpoint();
+    return rc;
   }
 
   res.state = ServerState::IN_PROGRESS;
-  return 0;
+  return RepStatus();
 }
 
 // Process get-status request
 // Kuaishou should create a new function for this request
-int CheckpointProducer::GetStatus(
+RepStatus CheckpointProducer::GetStatus(
                           const GetStatusRequest& req,
                           GetStatusResponse& res)
 {
   // We expect to get the same checkpoint_id as we provided in the CreateCheckpoint call
   if (req.checkpoint_id != checkpoint_id_) {
-    logger->Log(LogLevel::ERROR, FormatString("Invalid checkpoint id\n"));
-    return -1;
+    logger->Log(Severity::ERROR, FormatString("Invalid checkpoint id\n"));
+    DestroyCheckpoint();
```

Comment:
Why do we need to call DestroyCheckpoint here? I think it should be cleaned by the caller.

Reply:
You are right, it's redundant here. Removed.